code/modules/surgery/helpers.dm

```diff
@@ -0,0 +1,94 @@
+/proc/attempt_initiate_surgery(obj/item/I, mob/living/M, mob/user, var/override )
+	if(istype(M))
+		var/mob/living/carbon/human/H
+		var/obj/item/organ/external/affecting
+		var/selected_zone = user.zone_sel.selecting
+
+		if(istype(M, /mob/living/carbon/human))
+			H = M
+			affecting = H.get_organ(check_zone(selected_zone))
+
+		if(isslime(M) ||isalien(M) ||can_operate(H))	//if they're prone or a slime
```

Comment:
Err, this means one can do surgery on xenos and slimes while they are standing up, just anywhere, yes? I don't think the can_operate check requires them to be a human, I think you can get away with just carbon. After all, it seems you only need the affecting variable - use an `H` to call the get_organ proc, then use a carbon variable to check for operation ability.